src/awssdk_driver.rs

```diff
@@ -135,17 +134,23 @@ The following error may have more information: {}
 
 async fn build_client(env: &Environment) -> Result<QldbSessionSdk<DynConnector>> {
     let hyper = Client::https();
-    let provider = rusoto_driver::profile_provider(&env)?;
-    let rusoto_provider = match provider {
-        Some(p) => CredentialProvider::Profile(p),
-        None => CredentialProvider::Default(DefaultCredentialsProvider::new()?),
-    };
-    let creds = credentials::from_rusoto(rusoto_provider);
 
-    let conf = Config::builder()
-        .region(env.current_region())
-        .credentials_provider(creds);
+    let aws_config = aws_config::from_env();
+    let aws_config = match env.current_ledger().profile {
+        Some(ref name) => aws_config.credentials_provider(
+            LazyCachingCredentialsProvider::builder()
+                .load(
+                    ProfileFileCredentialsProvider::builder()
+                        .profile_name(name)
+                        .build(),
+                )
+                .build(),
+        ),
+        None => aws_config,
+    };
+    let aws_config = aws_config.load().await;
 
+    let conf = config::Builder::from(&aws_config).region(env.current_region());
```

Comment:
This is a little confusing (the way regions work), and I think will require another PR to cleanup. For now, the behavior is the same as before (which may/not be buggy, but definitely is hard to understand).

Reply:
you should be able to set the region on `aws_config` above. You may want to use `RegionProviderChain::first_try(env.current_region()).or_else_default()`

Reply:
Agreed with the direction, but there is some subtlety in the shell which lets people changes regions on the fly. I need to fix some of that madness first, which is sort of what I was alluding to.

Reply:
I'll link the next PR to this one so it's clear.